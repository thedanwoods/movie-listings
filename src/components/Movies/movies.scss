.movies__none {
  padding: 0 4px 16px;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 400;
  font-size: 13px;
  grid-column: 2 / 5;
}
