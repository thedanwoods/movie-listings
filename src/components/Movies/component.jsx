import React from 'react';
import PropTypes from 'prop-types';
import MovieCard from '../MovieCard';

import './movies.scss';

const byPopularityDescending = (a, b) => b.popularity - a.popularity;

const Movies = ({ movies }) => (
  <>
    {movies.length > 0 &&
      movies
        .sort(byPopularityDescending)
        .map(movie => <MovieCard key={movie.id} movie={movie} />)}
    {movies.length === 0 && (
      <div className="movies__none">
        No matching movies found.
      </div>
    )}
  </>
);

Movies.propTypes = {
  movies: PropTypes.array.isRequired,
};

export default Movies;
