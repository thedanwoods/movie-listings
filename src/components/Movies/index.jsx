import { connect } from 'react-redux';
import Movies from './component';
import getVisibleMovies from '../../utils/getVisibleMovies';

export const mapStateToProps = state => ({
  movies: getVisibleMovies(state.movies, state.genreFilter, state.ratingFilter),
});
export default connect(mapStateToProps)(Movies);
