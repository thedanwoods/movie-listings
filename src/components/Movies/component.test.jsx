import React from 'react';
import { shallow } from 'enzyme';
import Movies from './component';

const defaultProps = {
  movies: [
    {
      id: 1,
      popularity: 10,
    },
    {
      id: 2,
      popularity: 9,
    },
    {
      id: 3,
      popularity: 11,
    },
  ],
};

describe('<Movies /> component', () => {
  it('renders without errors', () => {
    const movies = shallow(<Movies {...defaultProps} />);

    expect(() => movies).not.toThrow();
    expect(movies.debug()).toMatchSnapshot();
  });

  it('displays none found message if empty movies array supplied', () => {
    const props = {
      movies: [],
    }
    const movies = shallow(<Movies {...props} />);

    expect(movies.find('.movies__none').exists()).toBe(true);
    expect(movies.debug()).toMatchSnapshot();
  });
});
