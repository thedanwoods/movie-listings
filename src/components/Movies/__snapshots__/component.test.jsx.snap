// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movies /> component displays none found message if empty movies array supplied 1`] = `
"<Fragment>
  <div className=\\"movies__none\\">
    No matching movies found.
  </div>
</Fragment>"
`;

exports[`<Movies /> component renders without errors 1`] = `
"<Fragment>
  <Connect(MovieCard) movie={{...}} />
  <Connect(MovieCard) movie={{...}} />
  <Connect(MovieCard) movie={{...}} />
</Fragment>"
`;
